serializers/src/glTF/2.0/babylon.glTFMaterial.ts

```diff
@@ -370,6 +385,40 @@ module BABYLON.GLTF2 {
             glTFMaterial.pbrMetallicRoughness = glTFPbrMetallicRoughness;
 
             materials.push(glTFMaterial);
+
+            return Promise.all(promises).then(() => {
+
+            });
+        }
+
+        /**
+         * 
+         * @param texture Texture with alpha to overwrite to one
+         * @param useAlpha Specifies if alpha should be preserved or not
+         * @returns Promise with texture
+         */
+        public static _SetAlphaToOneAsync(texture: BaseTexture, useAlpha: boolean): Promise<Texture> {
+            if (useAlpha) {
+                return Promise.resolve(texture as Texture);
+            }
+            else {
+                const scene = texture.getScene();
+                if (scene == null) {
+                    return Promise.reject(`Scene not available for texture ${texture.name}`);
+                }
+                else {
+                    const proceduralTexture = new ProceduralTexture('texture', texture.getSize(), 'setAlphaToOne', scene);

```

Comment:
Normally, you should wrap this like this`Promise.resolve().then(() => { ... })` to make sure any exceptions are captured by the promise, but if you end up using onLoadObservable, then just create a new Promise.

Reply:
ok, will do that

Reply:
@bghgary made the change

Reply:
You need to put the whole function inside the new Promise, not just the observable.